lib/Backend/GlobOpt.cpp

```diff
@@ -6935,20 +6960,34 @@ GlobOpt::SetSymStore(ValueInfo *valueInfo, Sym *sym)
     }
     if (valueInfo->GetSymStore() == nullptr || valueInfo->GetSymStore()->IsPropertySym())
     {
-        valueInfo->SetSymStore(sym);
+        SetSymStoreDirect(valueInfo, sym);
     }
 
     return sym;
 }
 
 void
+GlobOpt::SetSymStoreDirect(ValueInfo * valueInfo, Sym * sym)
+{
+    Sym * prevSymStore = valueInfo->GetSymStore();
+    if (prevSymStore && prevSymStore->IsStackSym() &&
+        prevSymStore->AsStackSym()->HasByteCodeRegSlot() &&
+        this->blockData.changedSyms)
```

Comment:
Will try remove it when propagating values across blocks, it should be fine if its construction is moved to InitBlockData.

---
In reply to: [68328835](https://github.com/Microsoft/ChakraCore/pull/1152#discussion_r68328835) [](ancestors = 68328835,68132431,68113857,68113107)